src/main/kotlin/dk/cachet/carp/webservices/data/service/DataStreamServiceImpl.kt

```diff
@@ -1,14 +1,33 @@
 package dk.cachet.carp.webservices.data.service
 
+import dk.cachet.carp.common.application.UUID
 import dk.cachet.carp.webservices.data.domain.DataStreamSequence
+import dk.cachet.carp.webservices.data.repository.DataStreamIdRepository
 import dk.cachet.carp.webservices.data.repository.DataStreamSequenceRepository
+import dk.cachet.carp.webservices.dataPoint.controller.DataPointController
+import kotlinx.datetime.Instant
+import kotlinx.datetime.toKotlinInstant
+import org.springframework.data.domain.PageRequest
+import org.springframework.stereotype.Service
 
+@Service
 class DataStreamServiceImpl(
-        private val dataStreamSequenceRepository: DataStreamSequenceRepository
+        private val dataStreamSequenceRepository: DataStreamSequenceRepository,
+        private val dataStreamIdRepository: DataStreamIdRepository
 ): IDataStreamService {
 
     override fun getDataStream(deploymentIds: List<String>): List<DataStreamSequence> {
         return dataStreamSequenceRepository.findAllByDeploymentIds(deploymentIds)
     }
 
+    override fun getLatestUpdatedAt(deploymentId: UUID): Instant?
+    {
+        val pageRequest = PageRequest.of(0, DataPointController.DEFAULT_PAGE_SIZE)
+        val dataStreamInputs = dataStreamIdRepository.getAllByDeploymentId(
+            deploymentId.stringRepresentation, pageRequest)
```

Comment:
Won't this only return DEFAULT_PAGE_SIZE amount of datastream ids? What if there is more? In that case the latest might not even be in it.

Reply:
Used wrong type of data structure - Page<DataStreamId> in query then of course I had to specify some DEFAULT_PAGE_SIZE.
Changed to List<>, so that's fixed, I suppose.